modules/programs/zsh.nix

```diff
@@ -272,6 +265,26 @@ in
 
         ${aliasesStr}
       '';
+    }
+    (mkIf (cfg.dotDir != null) {
+      home.sessionVariables.ZDOTDIR = zdotdir;
+    })
+    (mkIf (config.home.sessionVariableSetter == "zsh" && cfg.dotDir == null) {
+      home.file.".zshenv".text = ''
+        ${envVarsStr}
+      '';
+    })
+    (mkIf (config.home.sessionVariableSetter == "zsh" && cfg.dotDir != null) {
+      # When dotDir is set, only use ~/.zshenv to export ZDOTDIR, $ZDOTDIR/.zshenv for the rest
+      # This is so that if ZDOTDIR happens to be already set correctly (by e.g. spawning a zsh inside a zsh), all env vars still get exported
```

Comment:
Could you please wrap this comment to 80 characters as well? This line looks very long.